test/core/load_balancing/lb_policy_test_lib.h

```diff
@@ -329,30 +320,29 @@ class LoadBalancingPolicyTest : public ::testing::Test {
       // scheduled on the WorkSerializer.  We don't want to return until
       // all of those notifications have been delivered.
       absl::Notification notification;
-      test_->work_serializer_->Run(
-          [&]() ABSL_EXCLUSIVE_LOCKS_REQUIRED(*test_->work_serializer_) {
-            if (validate_state_transition) {
-              AssertValidConnectivityStateTransition(state_tracker_.state(),
-                                                     state, location);
-            }
-            LOG(INFO) << "Setting state on tracker";
-            state_tracker_.SetState(state, status, "set from test");
-            // SetState() enqueued the connectivity state notifications for
-            // the subchannel, so we add another callback to the queue to be
-            // executed after that state notifications has been delivered.
-            if (run_before_flush != nullptr) run_before_flush();
-            LOG(INFO) << "Waiting for state notifications to be delivered";
-            test_->work_serializer_->Run([&]() {
-              LOG(INFO) << "State notifications delivered, waiting for "
-                           "health notifications";
-              // Now the connectivity state notifications has been
-              // delivered. If the state reported was READY, then the
-              // pick_first leaf policy will have started a health watch, so
-              // we add another callback to the queue to be executed after
-              // the initial health watch notification has been delivered.
-              test_->work_serializer_->Run([&]() { notification.Notify(); });
-            });
-          });
+      test_->work_serializer_->Run([&]() {
```

Comment:
& seems very dangerous here... isn't this a bunch of reads of stack frames after we've returned?

Reply:
No, because the loop on line 346 below ensures that this will be executed before this function returns.

Also, this code isn't actually changing in this PR.  I only removed the thread annotation, which caused clang-format to reformat it.